libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt

```diff
@@ -275,13 +276,15 @@ internal class BillingController(
      * A route is considered the same if the waypoints count is the same and each pair of [`old waypoint` and `new waypoint`] are within [MAX_WAYPOINTS_DISTANCE_DIFF_METERS] of each other.
      * This method is also accounting for progress - if there's a multi-leg route, we'll only compare remaining legs of the current route against the new route.
      */
-    fun onExternalRouteSet(navigationRoute: NavigationRoute) {
+    fun onExternalRouteSet(navigationRoute: NavigationRoute, initialLegIndex: Int) {
         val runningSessionSkuId = getRunningOrPausedSessionSkuId()
         if (runningSessionSkuId == SessionSKUIdentifier.NAV2_SES_TRIP) {
             val currentRemainingWaypoints = getRemainingNonEVWaypointsOnRoute(
                 tripSession.getRouteProgress()
             )
-            val newWaypoints = getNonServerAddedWaypointsOnRoute(navigationRoute) { 1 }
+            val newWaypoints = getNonServerAddedWaypointsOnRoute(navigationRoute) {
+                countWaypointsUntilLegIndex(this, initialLegIndex) + 1
```

Comment:
@LukasPaczos could you please tell me if you see any red flags with this particular change?
TLDR: we should drop not only the origin of a new route, but all the waypoints up to the one that corresponds to initialLegIndex (if it's 0, it's the origin).

Reply:
I don't.